oximeter/db/src/native/io/column.rs

```diff
@@ -339,7 +359,16 @@ fn encode_value_array(values: ValueArray, mut dst: &mut BytesMut) {
             }
         }
         ValueArray::Ipv6(values) => copyout_pod_values!(Ipv6Addr, values, dst),
-        ValueArray::DateTime(values) => {
+        ValueArray::Date(values) => {
+            // Dates are represented in ClickHouse as a 16-bit unsigned number
+            // of days since the UNIX epoch.
+            dst.reserve(values.len() * std::mem::size_of::<u16>());
+            for value in values {
+                let days = value.signed_duration_since(EPOCH).num_days();
+                dst.put_u16_le(u16::try_from(days).unwrap());
```

Comment:
We shouldn't unwrap here. We can do that when deserializing values from ClickHouse, because the range must have already been validated to fit in a `u16` for ClickHouse to have stored it in the first place. These values are from _us_, and start as `chrono::NaiveDate`s, which have a wider range than ClickHouse supports. We need to check that and fail this call if they're out of range.

Reply:
Made this and the related pieces fallible in ff0af1775